cpp/models/lct_secir/parameters_io.h

```diff
@@ -450,17 +454,176 @@ IOResult<void> set_initial_data_from_confirmed_cases_impl(Populations& populatio
+*   equal to infectedCritical_reported after the function call if Group is set to zero in the beginning.
+* @returns Any io errors that happen during the scaling.
+*/
+template <class Populations, size_t Group = 0>
+IOResult<void> rescale_to_divi_data(Populations& populations, const ScalarType infectedCritical_reported,
+                                    const ScalarType infectedCritical_populations)
+{
+    if (floating_point_less<ScalarType>(infectedCritical_reported, 0., Limits<ScalarType>::zero_tolerance())) {
+        log_error("The provided reported number of InfectedCritical is negative. Please check the data.");
+        return failure(StatusCode::InvalidValue,
+                       "The provided reported number of InfectedCritical is negative. Please check the data.");
+    }
+
+    using LctStateGroup      = type_at_index_t<Group, typename Populations::LctStatesGroups>;
+    size_t first_index_group = populations.template get_first_index_of_group<Group>();
+
+    if (floating_point_equal<ScalarType>(infectedCritical_populations, 0., Limits<ScalarType>::zero_tolerance())) {
+        if (!(floating_point_equal<ScalarType>(infectedCritical_reported, 0., Limits<ScalarType>::zero_tolerance()))) {
+            log_info("The calculated number of patients in intensive care is zero, although the reported number is "
+                     "not. The reported number is uniformly distributed across groups and subcompartments. Note "
+                     "that this is not necessarily realistic.");
+            size_t num_InfectedCritical =
+                LctStateGroup::template get_num_subcompartments<InfectionState::InfectedCritical>();
+            ScalarType num_age_groups = (ScalarType)Populations::num_groups;
+            // Distribute reported number uniformly to age groups and subcompartments.
+            for (size_t subcompartment = 0;
+                 subcompartment < LctStateGroup::template get_num_subcompartments<InfectionState::InfectedCritical>();
+                 subcompartment++) {
+                populations[first_index_group +
+                            LctStateGroup::template get_first_index<InfectionState::InfectedCritical>() +
+                            subcompartment] =
+                    infectedCritical_reported / ((ScalarType)num_InfectedCritical * num_age_groups);
+            }
+            // Adjust Recovered compartment.
+            populations[first_index_group + LctStateGroup::template get_first_index<InfectionState::Recovered>()] -=
+                infectedCritical_reported / num_age_groups;
+            // Number of Susceptibles is not affected because Recovered is adjusted accordingly.
+        }
+    }
+    else {
+        // Adjust number of Recovered by adding the old number in InfectedCritical
+        // and subtracting the new number (= scaling_factor * old number).
+        ScalarType scaling_factor = infectedCritical_reported / infectedCritical_populations;
```

Comment:
```suggestion
        ScalarType scaling_factor_infectedCritical = infectedCritical_reported / infectedCritical_populations;
```
I would specify what this scaling factor is used for.